src/axom/sidre/core/ConduitMemory.cpp

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/sidre/core/ConduitMemory.hpp"
+
+namespace axom
+{
+namespace sidre
+{
+
+std::map<int, std::shared_ptr<ConduitMemory>> ConduitMemory::s_axomToInstance;
+std::map<conduit::index_t, std::shared_ptr<ConduitMemory>> ConduitMemory::s_conduitToInstance;
+const conduit::index_t ConduitMemory::s_defaultConduitId = conduit::Node().allocator();
+
+void ConduitMemory::privateRegisterAllocator()
+{
+  using conduit::utils::register_allocator;
+  auto deallocator = [](void* ptr) -> void {
+                       char* cPtr = (char*)(ptr);
+                       axom::deallocate<char>(cPtr);
+                     };
+  m_deallocCallback = deallocator;
+#if defined(AXOM_USE_CONDUIT_STD_FUNCTION)
+  m_allocCallback = [=](size_t itemCount, size_t itemByteSize) -> void* {
+                      void* ptr = axom::allocate<char>(itemCount * itemByteSize, m_axomId);
+                      return ptr;
+                    };
+  m_conduitId = register_allocator(m_allocCallback, m_deallocCallback);
+#else
+  /*
+    Note: conduit-0.9.4 allows the callbacks as std::function types.
+    Once we are there, we can use a single allocator, eliminating
+    the need for these if-else blocks.
+  */
+  if(m_axomId == MALLOC_ALLOCATOR_ID)
+  {
+    m_allocCallback = [](size_t itemCount, size_t itemByteSize) {
+                        void* ptr = axom::allocate<char>(itemCount * itemByteSize, MALLOC_ALLOCATOR_ID);
+                        return ptr;
+                      };
+    m_conduitId = register_allocator(m_allocCallback, m_deallocCallback);
+  }
+  else if(m_axomId == axom::INVALID_ALLOCATOR_ID)
+  {
+    m_allocCallback = nullptr;
+    m_conduitId = -1;
+  }
+  else if(m_axomId == 0)
```

Comment:
Would it be cleaner to restructure this and implement all these if-conditional checks in a while loop (i in {0, 20}) and a boolean, such as `found_it`, to break out of the loop and to use in the last else-clause?